src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -96,34 +96,53 @@ private void LoadingStatusIndicator_PropertyChanged(object sender, PropertyChang
             });
         }
 
-        // Indicates wether check boxes are enabled on packages
-        private bool _checkBoxesEnabled;
+        public bool CheckBoxesEnabled { get; set; }
 
-        public bool CheckBoxesEnabled
+        public bool IsSolution { get; set; }
+
+        public ObservableCollection<object> Items { get; } = new ObservableCollection<object>();
+        private ICollectionView CollectionView
         {
             get
             {
-                return _checkBoxesEnabled;
+                return CollectionViewSource.GetDefaultView(Items);
             }
-            set
+        }
+
+        /// <summary>
+        /// Count of Items (excluding Loading indicator) that are currently shown after applying any UI filtering.
+        /// </summary>
+        private int FilterCount
+        {
+            get
             {
-                _checkBoxesEnabled = value;
+                return PackageItemsFiltered.Count();
+            }
+        }
+
+        /// <summary>
+        /// All loaded Items (excluding Loading indicator) regardless of filtering.
+        /// </summary>
+        public IEnumerable<PackageItemListViewModel> PackageItems => Items.OfType<PackageItemListViewModel>().ToArray();
 
-                if (!_checkBoxesEnabled)
+        /// <summary>
+        /// Items (excluding Loading indicator) that are currently shown after applying any UI filtering.
+        /// </summary>
+        public IEnumerable<PackageItemListViewModel> PackageItemsFiltered
```

Comment:
left a comment below about confusion of the many types of filtering hapening here... i realized this is a 3rd type. Filtering the indicator from the rest of the packageitems.

Perhaps PackageItemsFiltered should be called PackageItems 
And PackageItems should be called PackageItemsView  (which adds the indicator?)

Reply:
We had: Items (raw list) and PackageItems (of type VM)
I added: FilteredItems and FilteredPackageItems (of type VM).

Can you explain the View in PackageItemsView?
Regarding overloading of the term "filtering". I am only using it one way: "UI filtering" being applied to the Items collection.
If you see other definitions, they were existing. For example "itemsFilter" is unintuitively meaning "the tab selected". I will gladly refactor that, preferably in a separate PR.

I've never seen old or new code that references the Loading indicator as being "filtered". Where do you see that?